docs/adr/0004-running-out-of-rancher-manager-cluster.md

```diff
@@ -0,0 +1,29 @@
+# 4. Running out of Rancher Manager cluster
+
+* Status: proposed
+* Date: 2023-08-31
+* Authors: @salasberryfin
+* Deciders: @richardcase @alexander-demicev @furkatgofurov7 @Danil-Grigorev @mjura
+
+## Context
+
+As an operator, I want the choice to deploy Rancher Turtles in the same or a different cluster to Rancher Manager, so that I have choice in my deployment topology.
+
+## Decision
+
+`CAPIImportReconciler` will contain two differentiated clients:
+- `Client`: in-cluster client for the controller and the CAPI resources.
+- `RancherClient`: for the Rancher Manager cluster.
+
+A flag `rancher-kubeconfig` with a path to a kubeconfig file is used to select the type of installation:
+- If no value is passed, Rancher Turtles is set to the same cluster installation: `Client` and `RancherClient` are the same instance of client.
+- If a valid path to a kubeconfig file is passed, `RancherClient` is set to client created from the kubeconfig.
+
+From an end-user perspective:
+
+* No parameter is required if Rancher Manager and Rancher Turtles run in the same cluster.
+* User will provide a path to a valid kubeconfig file that is available in the pod if installing Rancher Turtles outside of Rancher Manager cluster.
+
+## Consequences
+
+- The path to the kubeconfig must be mounted in the pod to be accessible.
```

Comment:
Might be worth adding that this is the pattern used by controller-runtime with the `--kubeconfig`